workflow-manager/limiter/limiter.go

```diff
@@ -0,0 +1,56 @@
+package limiter
+
+import "sync"
+
+// Limiter is similar to sync.Pool, however, it limits how many shared
+// objects (in this case workers) there can be
+type Limiter struct {
+	workers chan int32
+}
+
+// Ticket is the structure that allows you to schedule jobs
+type Ticket struct {
+	worker int32
+	lock   sync.Mutex
+	dirty  bool
+}
+
+// New creates a new instance of Limiter that limits how many functions can get executed without the other ones completing first
+func New(maxWorkers int32) *Limiter {
+	// Channels are blocking in go, if you write to them, something needs to read from them
+	// go allows us to create buffered channels, so that reading and writing from them is not blocking
+	// when the channel has items in the buffer, or has space in the buffer respectively
+	// using this feature of channels we can create a worker pool to execute tasks while limiting
+	// the resource creation that happens with each task
```

Comment:
Nitpicking this comment a bit: The comment spends four lines explaining how a Go feature works, but doesn't actually explain how this code takes advantage of the feature. The comment makes sense now, while you're learning about channels, but when you come back in three months you'll wonder why you were explaining channels. I would recommend linking to https://golang.org/ref/spec#Channel_types for the details, and spending these lines explaining the _how_ of this code.

An example of a _how_ comment:

We use a buffered channel of N elements, and fill it with N tokens. Before running, a task must read a token from the channel, and after running, a task must write a token back to the channel. Since there are only N tokens, we guarantee that any tasks trying to start while there are already N tasks running will block until one of those tasks returns its token.

(Also, I prefer capitalization and periods in all comments that are full sentences. Otherwise it's hard to tell where one sentence ends and the next begins).